### ISSUE TITLE
Discovery: review instances for vets-website 

### ISSUE THREAD

**Issue by rianfowler** [department-of-veterans-affairs/va.gov-team#5910]
# Overview 

- Devops team has rolled out [new approach](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/backend/review-instances.md) to doing review instances with a focus on editing code on the review instance. [product outline](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/platform/review-instances/product-outline.md)
- Would like to know how feasible it is for front end developers to use a review instance as a live coding environment 

# Notes

- Work with the devops team to get access to the current setup 
- Try to get vets-website live reloading on the review instance
  - Can potentially use https://github.com/webpack/webpack-dev-server#webpack-dev-server - there are some issues with this: 
  - How stable is it (webpack runs with everything in memory)? what happens if it crashes? 
  - what are the security concerns? 
- Focus primarily on identifying work. e.g. I think it will take a good chunk of work to make a robust watch task that has clear feedback
- Work with brooks / devops on understanding how the review instance status can be displayed on the PR. 
  - this part is a little murky. We spin up a review instance when a PR is updated, but do we update the review instance when changes are pushed? What happens if those changes are pushed from the review instance? 
- work closely with product on this- recommendations on how users can leverage this would be good 
  - how much friction is there now? how can we reduce it?
  - we don't have personas so for now think of how someone new vs someone experienced with the platform can use this?

**Comment by cvalarida** [comment]
> How stable is [the webpack-dev-server] (webpack runs with everything in memory)? what happens if it crashes?

I'm not sure how to answer this, but I imagine it'll be about as stable as it has been when we've run it locally. Don't know what other external...things...might affect them, but when we've run the dev server locally, it's pretty stable. As long as the code itself builds, anyhow.

The biggest concern I'd have is the amount of memory the dev server takes.

At any rate, I came here to mention [nodemon](https://nodemon.io) to handle it crashing. No idea yet if that would be beneficial, but there it is.

Potentially, we could:
1. Write a webpack-only build process
    - It would leave all the static content alone and just run webpack
    - We'll need to do this for splitting out the content once we move to a micro front end architecture anyhow
    - Level of effort: Low
2. Write a watch task that triggers a webpack-only build when a file is changed
    - I remember there being tools "out there" that do this, but I don't remember them off the top of my head
    - Level of effort: Unknown, but I suspect low

**Comment by meganhkelley** [comment]
@U-DON can you please summarize your findings from when you were looking at this a few months ago?

**Comment by U-DON** [comment]
I added a utility/command to run a local dev environment based on the review instance that's generated for a feature or development branch (assuming the developer has pushed their branch to GitHub).

Also wrote up a [guide for using it](https://github.com/department-of-veterans-affairs/va.gov-team/blob/6afeeb3fec73d3ed4013730698167f37a98e6dfb/platform/engineering/frontend/vets-website/local-development-with-review-instances.md).

The finding was that it's possible to run local builds with other APIs, including review instance APIs. The only issue is that logins will have unintended effects with redirects, so I've included some workarounds for it in the guide.
